unlock-app/src/components/helpers/Balance.js

```diff
@@ -3,50 +3,84 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import Web3Utils from 'web3-utils'
 
+import UnlockPropTypes from '../../propTypes'
+import { formatEth, formatCurrency } from '../../selectors/currency'
+
 /**
  * Component which shows a balance in Eth
  * @param {*} amount: the amount to convert to Eth
  * @param {string} unit: the unit of the amount to convert to Eth
+ * @param {object} conversion: a hash of conversion values for ether to currencies
+ * @param {function} EthComponent: a React component that displays an ether value
  */
-export function Balance({ amount, unit = 'wei' }) {
-  let inWei = Web3Utils.toWei(amount || '0', unit)
-  let inEth = Web3Utils.fromWei(inWei, 'ether')
-  let ethWithPresentation = BalancePresenter(inEth)
-
-  return (<BalanceWithUnit>
-    三 
-    {' '}
-    {ethWithPresentation}
-  </BalanceWithUnit>)
-}
-
-/**
- * Provide a representaion of Eth Balances suitable for the Unlock frontend
- * application.
- * @param {string} eth: An amount of Eth
- */
-function BalancePresenter(eth){
-  const DECIMAL_PLACES = 2
-  const SIGNIFICANT_DIGITS = 2
-  const MINIMUM_THRESHOLD = 0.0001
-
-  let numericalEth = Number(eth)
-
-  switch(true) {
-  case numericalEth < MINIMUM_THRESHOLD && numericalEth > 0 :
-    return '< 0.0001'
-  case numericalEth < 1:
-    return parseFloat(numericalEth.toPrecision(SIGNIFICANT_DIGITS))
-  default:
-    return numericalEth.toFixed(DECIMAL_PLACES)
+export function Balance({ amount, unit = 'wei', conversion = { USD: undefined }, EthComponent = ({ value }) => value }) {
+  let inEth
+  if (unit !== 'dollars' && unit !== 'eth') {
+    const inWei = Web3Utils.toWei(amount || '0', unit)
+    inEth = Web3Utils.fromWei(inWei, 'ether')
+  } else {
+    inEth = +amount
```

Comment:
Isn't that happening either when unit is `dollars` or `eth`?

Reply:
amount is usually a string, formatEth expects a number

Reply:
oh yes of course, but I meant that `inEth` is probably a poor choice for a value which could be actually dollars?

Reply:
I see. I renamed it to currency, hopefully that resolves this :)